### PR TITLE
Fix "key count" setting showing as "circle size" in osu!mania editor

### LINKED ISSUE #12904: Select key count of a mania map in the lazer editor
Option to configure key count (to set 4k, 5k, 7k, for example) in the setup page for mania mode of osu!lazer edtior. Currently the lazer editor only support 5k mapping for mania mode (correct me if I am wrong).

Just in my opinion this is truely an important feature, to let mappers get the key count they want. I have searched the web to find if there is already a option doing this, ending up only found [the Reddit answer](https://www.reddit.com/r/osugame/comments/kagjoy/how_to_change_amount_of_keys_used_for_an_osumania/gfdurcw/) .

### PR DESCRIPTION
**Pull request by peppy**
Closes https://github.com/ppy/osu/issues/12904.

### CODE REVIEW

File: osu.Game/Screens/Edit/Setup/DifficultySection.cs [hunk 1]
```diff
@@ -13,14 +13,14 @@
...
-        private LabelledSliderBar<float> approachRateSlider = null!;
-        private LabelledSliderBar<float> overallDifficultySlider = null!;
-        private LabelledSliderBar<double> baseVelocitySlider = null!;
-        private LabelledSliderBar<double> tickRateSlider = null!;
+        protected LabelledSliderBar<float> CircleSizeSlider = null!;
+        protected LabelledSliderBar<float> HealthDrainSlider = null!;
+        protected LabelledSliderBar<float> ApproachRateSlider = null!;
+        protected LabelledSliderBar<float> OverallDifficultySlider = null!;
+        protected LabelledSliderBar<double> BaseVelocitySlider = null!;
+        protected LabelledSliderBar<double> TickRateSlider = null!;
```
Comment (bdach) [inline osu.Game/Screens/Edit/Setup/DifficultySection.cs:23]: Should probably be `readonly`

File: osu.Game.Rulesets.Mania/Edit/Setup/ManiaDifficultySection.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+using osu.Game.Screens.Edit.Setup;
+
+namespace osu.Game.Rulesets.Mania.Edit.Setup
+{
+    public partial class ManiaDifficultySection : DifficultySection
+    {
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            CircleSizeSlider.Label = BeatmapsetsStrings.ShowStatsCsMania;
```
Comment (bdach) [inline osu.Game.Rulesets.Mania/Edit/Setup/ManiaDifficultySection.cs:15]: This is fine and all, but if we're doing this, we should probably also hide the AR slider in mania, and both the CS and AR sliders in taiko, as they do absolutely nothing. And rename base velocity in taiko to scroll speed.

File: osu.Game.Rulesets.Mania/Edit/Setup/ManiaDifficultySection.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+            CircleSizeSlider.Description = "The number of columns in the beatmap";
```
Comment (bdach) [inline osu.Game.Rulesets.Mania/Edit/Setup/ManiaDifficultySection.cs:16]: Not something to address in this pull, but the "fun" thing with this is that the base `DifficultySection` has localisable descriptions, but we can't actually do that in ruleset DLLs right now... Going to need tooling support.

**Comment by bdach** [comment]
I also wouldn't go as far as to say that this closes the linked issue because the slider is still half broken (requires save and full editor reload to take effect).

File: osu.Game.Rulesets.Mania/Edit/Setup/ManiaDifficultySection.cs (hunk 2 above)
Comment (peppy) [reply osu.Game.Rulesets.Mania/Edit/Setup/ManiaDifficultySection.cs:15]: Yeah, I have a feeling there would be others, which is why I made the implementation allow full control. I'll fix the precision here, but the others should probably be added in a separate pull?

File: osu.Game.Rulesets.Mania/Edit/Setup/ManiaDifficultySection.cs (hunk 3 above)
Comment (peppy) [reply osu.Game.Rulesets.Mania/Edit/Setup/ManiaDifficultySection.cs:16]: Yep.

File: osu.Game.Rulesets.Mania/Edit/Setup/ManiaDifficultySection.cs (hunk 2 above)
Comment (bdach) [reply osu.Game.Rulesets.Mania/Edit/Setup/ManiaDifficultySection.cs:15]: Sure, we can do the rest separately.

**Review by bdach — APPROVED** [review]

**Pull request merged**